config/confighttp/confighttp.go

```diff
@@ -77,21 +77,20 @@ type ClientConfig struct {
 	CompressionParams configcompression.CompressionParams `mapstructure:"compression_params"`
 
 	// MaxIdleConns is used to set a limit to the maximum idle HTTP connections the client can keep open.
-	// By default, it is set to 100.
-	MaxIdleConns *int `mapstructure:"max_idle_conns"`
+	// By default, it is set to 100. Zero means no limit.
```

Comment:
Based on the change at line 163, it looks like it's not possible to set this to 0